File: Recipes/Recipes - backend/Recipes/Recipes/Services/Service/RecipeService.cs
using AutoMapper;
using Recipes.DTOs;
using Recipes.DTOs.Delete;
using Recipes.DTOs.Get.Recipe;
using Recipes.DTOs.Post.Recipe;
using Recipes.Models;
using Recipes.Models.Ingredient;
using Recipes.Models.Users;
using Recipes.Repositorys.Interface;
using Recipes.Services.Interface;

namespace Recipes.Services.Service
{
    public class RecipeService:IRecipeService
    {
        public readonly IRecipeRepository _recipes;
        public readonly IMapper _mapper;

        public RecipeService(IRecipeRepository recipes, IMapper mapper)
        {
            _recipes = recipes;
            _mapper = mapper;
        }

        public RecipePostResponse createRecipe(RecipePostRequest recipe,string username)
        {
            //_mapper.Map<IngredientWithQuantity>(recipe.Ingredients);
            var response = _mapper.Map<RecipePostResponse>(_recipes.createRecipe(_mapper.Map<Recipe>(recipe),username));
            if(response == null)
            {
                return null;
            }
            return response;
        }

        public RecipeDeleteResponse deleteRecipe(string name, string username)
        {
            var response = _mapper.Map<RecipeDeleteResponse>(_recipes.deleteRecipe(name, username));
            if (response == null)
            {
                return null;
            }
            return response;
        }

        public List<RecipesGetResponse> getAllRecipes()
        {
            var response = _mapper.Map<List<RecipesGetResponse>>(_recipes.getAllRecipes());
            if (response == null)
            {
                return null;
            }
            return response;
        }

        public List<RecipesGetResponse> getSearchedRecipes(string name, string ingredientname)
        {
            var response = _mapper.Map<List<RecipesGetResponse>>(_recipes.getSearchedRecipes(name,ingredientname));
            if (response == null)
            {
                return null;
            }
            return response;
        }
    }
}

File: Recipes/Recipes - frontend/src/services/recipe.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, Input } from '@angular/core';
import { Router } from '@angular/router';
import { Observable, map } from 'rxjs';
import { IngredientDTO } from 'src/models/IngredientDTO.model';
import { PostRecipe } from 'src/models/PostRecipe.model';
import { Recipe } from 'src/models/Recipe.model';
import { RecipeSearch } from 'src/models/RecipeSearch.model';

@Injectable({
  providedIn: 'root'
})
export class RecipeService {

  rootUrl:string = 'http://localhost:5121/api/Recipe/';

  constructor(private http: HttpClient,public router:Router) { }

  getAllRecipes() {
    let recipesUrl = this.rootUrl + 'get-all-recipes';
    return this.http.get<Recipe[]>(recipesUrl);
  }

  createRecipe(recipe: PostRecipe) : Observable<PostRecipe> {
    var url = this.rootUrl + 'create-recipe';
    var body = recipe;

    return this.http.post<any>(url, body).pipe(
      map(response => {
        console.log(response);
        return new PostRecipe(response.name,response.ingredients,response.description);
      })
    )
  }

  deleteRecipe(recipe: string) : Observable<IngredientDTO> {
    var url = this.rootUrl + "delete-recipe?name=" + recipe;

    return this.http.delete<IngredientDTO>(url);
  }

  getSearchedRecipes(recipe:RecipeSearch){
    var url = "http://localhost:5121/api/Recipe/get-searched-recipes?name=" + recipe.name + "&ingredientname=" + recipe.ingredientSeach;
    return this.http.get<Recipe[]>(url);
  }
}
